galette/lib/Galette/Entity/Adherent.php

```diff
@@ -857,9 +858,60 @@ public function getDues(): string
                 $ret = _T("No longer member");
             }
         }
+
+        if (self::getCountDonations() > 0) {
+            if ($never_contributed) {
+                $ret = _T("Never contributed, but a donation");
+            } else {
+                $ret .= ' - '._T("Donator");
+            }
+        }
         return $ret;
     }
 
+    /**
+     * Get donation count in the current year
+     *
+     * @return int count 0..x
+     */
+    public function getCountDonations()
+    {
+        global $preferences; //$this->preferences n'est pas initialisée ? on retrouve la même ligne dans validate, canXXX()
```

Comment:
It is only set from `Adherent::setDependencies()` method, but it's currently not set per default because it's not useful everywhere.
New DI system should be used to avoid that, but changing everything is a huge work; so we keep going with globals right now for those cases.